src/main/java/de/zalando/aruha/nakadi/exceptions/NakadiException.java

```diff
@@ -26,4 +31,11 @@ public void setProblemMessage(final String problemMessage) {
         this.problemMessage = problemMessage;
     }
 
+    public Problem asProblem() {
+        final String detail = StringUtils.isNotBlank(problemMessage) ? problemMessage : getMessage();
+        return Problem.valueOf(getStatus(), detail);
+    }
+
+    protected abstract Response.StatusType getStatus();
```

Comment:
I think that it's going to work just fine. The only scenario that it would not work is that if we have different problems for the same exception, depending on the context. Then we would be forced to create a new Exception, just to have a different status code.